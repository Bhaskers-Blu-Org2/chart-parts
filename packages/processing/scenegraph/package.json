{
	"name": "@markable/scenegraph",
	"description": "A library that parses Vega scenegraphs into an object model",
	"version": "0.0.15",
	"main": "lib/index.js",
	"module": "lib/index.js",
	"types": "lib/index.d.ts",
	"license": "MIT",
	"scripts": {
		"clean": "rimraf lib",
		"start": "tsc -w --preserveWatchOutput",
		"build": "tsc",
		"lint": "tslint src/**/*.ts*"
	},
	"devDependencies": {
		"rimraf": "^2.6.2",
		"typescript": "^3.0.1",
		"tslint": "^5.11.0"
	},
	"dependencies": {
		"@markable/interfaces": "^0.0.15"
	}
}
