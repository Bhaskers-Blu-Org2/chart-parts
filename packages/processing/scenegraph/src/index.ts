export { registerItemType } from './registry'
export * from './parse'
