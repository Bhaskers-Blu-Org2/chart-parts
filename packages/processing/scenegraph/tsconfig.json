{
	"extends": "../../../tsconfig.json",
	"compilerOptions": {
		"lib": ["esnext", "dom"],
		"outDir": "lib",
		"composite": true,
		"rootDir": "./src",
		"baseUrl": "./src",
		"tsBuildInfoFile": "./lib/.tsbuildinfo"
	},
	"files": ["src/index.ts"],
	"include": ["./src/**/*.ts*"]
}
