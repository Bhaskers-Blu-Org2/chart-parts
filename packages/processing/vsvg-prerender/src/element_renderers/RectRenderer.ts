import { Path } from 'd3-path'
import { SGMark, SGRectItem, MarkType } from '@gog/mark-interfaces'
import { VSvgNode } from '@gog/vdom-interfaces'
import { copyCommonProps, assertTypeIs, emitMarkGroup } from './util'
import { rectangle } from '../path'
import { MarkPrerenderer } from '@gog/prerender-interfaces'
import { VSvgMarkPrerenderer } from './interfaces'

export class RectRenderer implements VSvgMarkPrerenderer {
	public static TARGET_MARK_TYPE = MarkType.Rect

	public render(mark: SGMark<SGRectItem>) {
		assertTypeIs(mark, RectRenderer.TARGET_MARK_TYPE)

		const nodes = emitMarkGroup(
			MarkType.Rect,
			mark.role,
			mark.items.map(item => {
				const { x = 0, y = 0 } = item
				const result: VSvgNode = {
					type: 'path',
					attrs: {
						d: rectangle(item, x, y).toString(),
					},
				}
				copyCommonProps(item, result)
				return result
			}),
		)
		return { nodes }
	}
}
