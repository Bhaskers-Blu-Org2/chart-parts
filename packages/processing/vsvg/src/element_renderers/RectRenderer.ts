import {
	MarkType,
	SGMark,
	SGRectItem,
	getItemSpace,
	VSvgNode,
} from '@gog/interfaces'
import { commonProps, assertTypeIs, emitMarkGroup } from './util'
import { rectangle } from '../path'
import { VSvgMarkConverter } from './interfaces'

declare var require: any
// tslint:disable-next-line no-var-requires no-submodule-imports
const assign = require('lodash/assign')

export class RectRenderer implements VSvgMarkConverter {
	public static TARGET_MARK_TYPE = MarkType.Rect

	public render(mark: SGMark<SGRectItem>) {
		assertTypeIs(mark, RectRenderer.TARGET_MARK_TYPE)

		const nodes = emitMarkGroup(
			MarkType.Rect,
			mark.role,
			mark.items.map(item => {
				const space = getItemSpace(item)
				const result: VSvgNode = {
					type: 'path',
					attrs: assign({}, commonProps(item), {
						d: rectangle(
							assign({}, item, space.shape),
							space.origin.x,
							space.origin.y,
						).toString(),
					}),
					metadata: item.metadata,
					channels: item.channels,
				}
				return result
			}),
		)
		return { nodes }
	}
}
