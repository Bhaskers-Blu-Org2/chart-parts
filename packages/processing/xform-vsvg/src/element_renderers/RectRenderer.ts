import { Path } from 'd3-path'
import { MarkType } from '@gog/mark-interfaces'
import { SGMark, SGRectItem } from '@gog/scenegraph-interfaces'
import { VSvgNode } from '@gog/vdom-interfaces'
import { commonProps, assertTypeIs, emitMarkGroup } from './util'
import { rectangle } from '../path'
import { VSvgMarkConverter } from './interfaces'

export class RectRenderer implements VSvgMarkConverter {
	public static TARGET_MARK_TYPE = MarkType.Rect

	public render(mark: SGMark<SGRectItem>) {
		assertTypeIs(mark, RectRenderer.TARGET_MARK_TYPE)

		const nodes = emitMarkGroup(
			MarkType.Rect,
			mark.role,
			mark.items.map(item => {
				const { x = 0, y = 0, x2 = 0, y2 = 0 } = item
				const height = item.height || y2 - y
				const width = item.width || x2 - x

				const result: VSvgNode = {
					type: 'path',
					attrs: {
						...commonProps(item),
						d: rectangle({ ...item, width, height }, x, y).toString(),
					},
					metadata: item.metadata,
					channels: item.channels,
				}
				return result
			}),
		)
		return { nodes }
	}
}
