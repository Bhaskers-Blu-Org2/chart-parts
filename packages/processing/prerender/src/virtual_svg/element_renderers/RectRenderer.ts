import { Path } from 'd3-path'
import { Mark, RectItem } from '@gog/scenegraph'
import { VSvgNode } from '@gog/vdom-interfaces'
import { copyCommonProps, assertTypeIs } from './util'
import { rectangle } from '../path/shapes'
import { MarkPrerenderer } from '../../interfaces'

export class RectRenderer implements MarkPrerenderer<VSvgNode[]> {
	public render(mark: Mark) {
		assertTypeIs(mark, RectItem.ITEM_TYPE)

		// Render each item embedded in this mark
		const renderedItems: VSvgNode[] = mark.items.map((item: RectItem) => {
			const rendered = rectangle(item)
			const result: VSvgNode = {
				type: 'path',
				attrs: { d: rendered.toString() },
			}
			copyCommonProps(item, result)
			return result
		})

		return renderedItems
	}
}
