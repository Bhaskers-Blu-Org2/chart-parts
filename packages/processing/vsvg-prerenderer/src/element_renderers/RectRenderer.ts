import { Path } from 'd3-path'
import { Mark, RectItem } from '@gog/scenegraph'
import { VSvgNode } from '@gog/vdom-interfaces'
import { copyCommonProps, assertTypeIs } from './util'
import { rectangle } from '../path'
import { MarkPrerenderer } from '@gog/prerender-interfaces'

export class RectRenderer implements MarkPrerenderer<VSvgNode[]> {
	public static TARGET_MARK_TYPE = RectItem.ITEM_TYPE

	public render(mark: Mark<RectItem>) {
		assertTypeIs(mark, RectRenderer.TARGET_MARK_TYPE)

		// Render each item embedded in this mark
		const renderedItems: VSvgNode[] = mark.items.map(item => {
			const result: VSvgNode = {
				type: 'path',
				attrs: { d: rectangle(item).toString() },
			}
			copyCommonProps(item, result)
			return result
		})

		return renderedItems
	}
}
