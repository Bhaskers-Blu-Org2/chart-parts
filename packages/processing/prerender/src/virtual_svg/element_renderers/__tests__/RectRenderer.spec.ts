import { Mark, RectItem } from '@gog/scenegraph'
import { RectRenderer } from '../RectRenderer'

describe('The Rect Rendered', () => {
	it('can render a rectangle', () => {
		const mark = new Mark()
		mark.marktype = RectItem.ITEM_TYPE
		const rect = new RectItem()
		rect.x = 0
		rect.y = 0
		rect.width = 100
		rect.height = 100
		rect.fill = 'red'
		rect.stroke = 'blue'
		mark.items.push(rect)

		const renderer = new RectRenderer()
		const result = renderer.render(mark)

		const output = document.createElement('svg')
		output.setAttribute('width', '250')
		output.setAttribute('height', '250')

		const markElement = document.createElement(result[0].element)
		Object.keys(result[0].attrs).map(attr =>
			markElement.setAttribute(attr, result[0].attrs[attr]),
		)
		output.appendChild(markElement)

		expect(output.outerHTML).toMatchSnapshot()
	})
})
