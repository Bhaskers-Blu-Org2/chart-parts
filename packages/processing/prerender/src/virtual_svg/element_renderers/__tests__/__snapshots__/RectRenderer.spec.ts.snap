// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Rect Rendered can render a rectangle 1`] = `
<div>
  <svg
    height="250"
    width="250"
  >
    <path
      d="M0,0h100v100h-100Z"
    />
  </svg>
</div>
`;
